src/realm/object-store/impl/epoll/external_commit_helper.cpp

```diff
@@ -306,19 +252,87 @@ void ExternalCommitHelper::DaemonThread::listen()
             continue;
         }
 
-        if (ev.data.u32 == (uint32_t)m_shutdown_read_fd) {
+        if (ev.data.ptr == this) {
+            // Shutdown fd was notified, so exit
             return;
         }
 
+        // One of the ExternalCommitHelper fds was notified. We need to check
+        // if the target is still alive while holding the mutex, but we need to
```

Comment:
there are two mutexes involved here but this comment just refers to "the mutex" - the way i read this code is that there are two mutexes, we use one to protect the list of live coordinators and one to serialize access to calling `on_change()`. is that correct? i assume so from the comment around the mutex declarations, but just checking.

Reply:
Not quite. The second mutex is just to make it so that `remove()` can wait for the current call to `on_change()` to complete, which is required for `Realm::close()` to work as desired. I forgot to update this comment when introducing that mutex.